src/matrix_common/versionstring.py

```diff
@@ -44,6 +47,11 @@ def get_distribution_version_string(distribution_name: str) -> str:
     Args:
         distribution_name: The name of the distribution package to check the version of
 
+        cwd: if provided, the directory to run all git commands in. `cwd` may also be
+            the path to a file, in which case `os.path.dirname(cwd)` is used instead.
```

Comment:
I _definitely_ had code which did this. Must have lost it when force pushing. Will fix on main branch and do another release.